### PR TITLE
Refactor analysis vi

### PR DESCRIPTION
**Pull request by mschwoer**
complete the analysis refactoring by simplifying & (re)moving code

add new feature: download plotting parameters and show them on the "results" page

### CODE REVIEW

File: alphastats/gui/pages/06_Results.py [hunk 1]
```diff
@@ -21,9 +21,11 @@
 
     method = saved_item[0]
```
Comment (boopthesnoot) [inline alphastats/gui/pages/06_Results.py:22]:
```suggestion
for count, (method, plot, parameters) in enumerate(st.session_state[StateKeys.PLOT_LIST]):
```
we could change the loop to this, or is it making this less readable?

File: alphastats/gui/utils/analysis.py [hunk 2]
```diff
@@ -58,22 +46,22 @@ def show_widget(self):  # noqa: B027
         pass
 
     def do_analysis(self):
-        """Perform the analysis.
+        """Perform the analysis after an optional check for NaNs.
```
Comment (boopthesnoot) [inline alphastats/gui/utils/analysis.py:49]: i didn't understand what you meant by saying that this check is optional?

File: alphastats/gui/utils/analysis.py [hunk 3]
```diff
@@ -449,4 +448,20 @@ def _do_analysis(self):
             covar=self._parameters["covar"],
             between=self._parameters["between"],
         )
-        return ancova_analysis, None, self._parameters
+        return ancova_analysis, None
+
+
+ANALYSIS_OPTIONS = {
```
Comment (boopthesnoot) [inline alphastats/gui/utils/analysis.py:454]: would moving the constants to the beggining of the file make more sense?

**Review by boopthesnoot — APPROVED** [review]
LGTM

File: alphastats/gui/pages/06_Results.py (hunk 1 above)
Comment (mschwoer) [reply alphastats/gui/pages/06_Results.py:22]: in principle, yes, but then the `.remove(saved_item)` would become less readable

File: alphastats/gui/utils/analysis.py (hunk 2 above)
Comment (mschwoer) [reply alphastats/gui/utils/analysis.py:49]: will rephrase it to `Perform the analysis after some upfront method-dependent checks (e.g. or NaNs)`

File: alphastats/gui/utils/analysis.py (hunk 3 above)
Comment (mschwoer) [reply alphastats/gui/utils/analysis.py:454]: totally, but this needs to be done after the class definitions referenced in the dict
